frontend/alert/src/components/ManageAlert.js

```diff
@@ -0,0 +1,348 @@
+                                 <Img src={ABell} width="0.5rem" height="0.5rem" />
+                                 <HeaderAction>RESUBSCRIBE</HeaderAction>
+                             </Flex>
+                             <Flex valign className={styles.headerbuttons}>
+                                 <Img src={XBell} width="0.5rem" height="0.5rem" />
+                                 <HeaderAction>CANCEL</HeaderAction>
+                             </Flex>
+                             <Flex valign className={styles.headerbuttons}>
+                                 <Img src={Trash} width="0.5rem" height="0.5rem" />
+                                 <HeaderAction>DELETE</HeaderAction>
+                             </Flex>
+                         </RightItem>
+                     </GridItem>
+                 </>
+             )}
+
+        </>
+    );
+};
+
+Header.propTypes = { selected: PropTypes.number };
+
+export const ManageAlertHeader = () => (
+    <Flex margin="-3.4rem 0rem 0rem 0rem">
+        <img
+            alt="Penn Course Alert logo"
+            src={Logo}
+            width="50rem"
+        />
+        <Flex col center valign halign margin="1.5rem 1.5rem 1.5rem 2rem" padding="2.2rem 0rem 0rem 0rem">
+            <Flex>
+                <Input placeholder="Course" />
+                <Button>Alert me</Button>
+            </Flex>
+            <P size="0.9rem" margin="1rem 0rem 0rem -3rem">Alert me until I cancel</P>
+        </Flex>
+    </Flex>
+);
+
+const AlertSearch = () => (
+    <SearchFlex valign>
+        <Flex valign margin="0.2rem" className={styles.search}>
+            <Img src={Search} alt="" width="0.6rem" height="0.6rem" />
+            <SearchInput type="search" placeholder="Search" />
+        </Flex>
+    </SearchFlex>
+);
+
+const AlertAction = Object.freeze({ Resubscribe: 1, Cancel: 2 });
+const AlertStatus = Object.freeze({ Closed: 1, Open: 2 });
+const AlertRepeat = Object.freeze({ Inactive: 1, EOS: 2, Once: 3 });
+
+const ActionButton = ({ type }) => {
+    let img;
+    let primary;
+    let secondary;
+    let text;
+
+    switch (type) {
+        case 1:
+            primary = "#5891fc";
+            secondary = "rgba(88, 145, 252, 0.12)";
+            img = Bell;
+            text = "Resubscribe";
+            break;
+        case 2:
+            primary = "#646e7a";
+            secondary = "rgba(162, 169, 176, 0.15)";
+            img = XBell;
+            text = "Cancel";
+            break;
+        default:
+    }
+
+    return (
+        <ActionFlex valign halign background={secondary}>
+            <Flex valign margin="0.3rem" className={styles.actionbutton}>
+                <P size="0.6rem" color={primary} weight="600">{text}</P>
+                <Img src={img} width="0.6rem" height="0.6rem" alt="" />
+            </Flex>
+        </ActionFlex>
+    );
+};
+
+ActionButton.propTypes = { type: PropTypes.oneOf([1, 2]) };
+
+
+const AlertItem = ({
```

Comment:
I think it would be great for these larger components to get some explanatory comments about what they are/what they do. Maybe even split them out into their own file, just to make the file sizes a bit more manageable?

Reply:
Done!